components/statsTracking/fn_addStatsEventHandlersToClass.sqf

```diff
@@ -0,0 +1,44 @@
+// Adds event handlers to the given unit, to track various stats for viewing on the debriefing.
+// These event handlers are local to the client, and the data they collect is relevant only 
+// to the client, so we don't need to sync the data across the network.
+params ["_unit"];
+
+if !(_unit isEqualTo player) exitWith {};
+
+if (hasInterface) then {
+	["ace_firedPlayer", {
+		params ["_unit", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile"];
+		if (_unit == player) then {
+			_currentShots = missionNamespace getVariable ["cafe_playerShots", createHashMap];
+			_currentCount = _currentShots getOrDefault [_ammo, 0];
+			_updated = _currentCount + 1;
+			_currentShots set [_ammo, _updated];
+			missionNamespace setVariable ["cafe_playerShots", _currentShots, false];
+		};
+	}] call CBA_fnc_addEventHandler;
+
+	["ace_unconscious", {
+		params ["_unit", "_state"];
+
+		if (_unit == player && _state == true) then {
+			_currentUncons = missionNamespace getVariable ["cafe_playerUncons", 0];
+			_updated = _currentUncons + 1;
+			missionNamespace setVariable ["cafe_playerUncons", _updated, false];
+		};
+	}] call CBA_fnc_addEventHandler;
+
+	addMissionEventHandler ["Ended",
+	{
+		systemChat "Ending";
+		_currentShots = missionNamespace getVariable ["cafe_playerShots", createHashMap];
+		_uncons = missionNamespace getVariable ["cafe_playerUncons", 0];
+		_text = "";
+		{
+			_ammo = _x;
+			_shots = _y;
+			_text = _text + format ["%1 | Fired: %2<br/>", _ammo, _shots];
+		} forEach _currentShots;
+		_text = _text + format ["You went unconscious %1 times<br/>", _uncons];
```

Comment:
does SQF have a nice "pluralise" function which could sort out the "You went unconscious 1 times" annoyance?

Reply:
Nope, closest is to use `select` with a boolean if you want to do it inline:
https://community.bistudio.com/wiki/select

Reply:
Alas. I'll just swap it for `time(s)` instead I guess.